azure-resources/src/main/java/io/opentelemetry/contrib/azure/resource/AzureResourceDetector.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.azure.resource;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.resources.ResourceBuilder;
+
+public class AzureResourceDetector implements ComponentProvider<Resource> {
+
+  @Override
+  public Class<Resource> getType() {
+    return Resource.class;
+  }
+
+  @Override
+  public String getName() {
+    return "azure";
+  }
+
+  @Override
+  public Resource create(DeclarativeConfigProperties config) {
```

Comment:
If it ever makes sense for these to be configurable, declarative config supports passing configuration to resource detectors